operators/utilities.py

```diff
@@ -38,6 +38,52 @@ def select_all(context: Context):
 def deselect_all(context: Context):
     global_data.selected.clear()
 
+def select_invert(context: Context):
+    sketch = context.scene.sketcher.active_sketch
+    if sketch:
+        generator = sketch.sketch_entities(context)
+    else:
+        generator = entities_3d(context)
+    
+    for e in generator:
+        e.selected = not e.selected
+
+def select_extend(context: Context):
+    sketch = context.scene.sketcher.active_sketch
+    if sketch:
+        generator = sketch.sketch_entities(context)
+    else:
+        generator = entities_3d(context)
+    
+    to_select = []
+    for e in generator:
+        if e.is_segment():
+            if e.selected:
+                if hasattr(e, 'p1'):
+                    to_select.append(e.p1)
+                if hasattr(e, 'p2'):
+                    to_select.append(e.p2)
+                continue
+            if (hasattr(e,'p1') and e.p1.selected) or (hasattr(e, 'p2') and e.p2.selected):
+                to_select.append(e)    
+                continue
+    
+    for coindident in context.scene.sketcher.constraints.coincident:
+        for entity in coindident.entities():
+            if entity.selected:
+                for entity_to_select in coindident.entities():
+                    to_select.append(entity_to_select)
+                continue    
+            
+    is_something_to_select = False
+    for entity in to_select:
+        if not entity.selected:
+            is_something_to_select = True
+            entity.selected = True
+    
+    return is_something_to_select
```

Comment:
Slight simplification:
```py
is_something_to_select = not all(e.selected for e in to_select)
for entity in to_select:
    entity.selected = True
```

Reply:
Secondly, why would you return whether something has been selected or not?

Reply:
if you this return value does not matter, I would suggest to remove the `to_select` variable and simply write `entity.selected = True` where applicable.

Reply:
i'm looping until select_extend return False (in select.py). This is needed for extend sleection with all in chain.

Reply:
i Simplified this as @amrsoll sugested 👍